server/server.go

```diff
@@ -676,6 +678,83 @@ func startMetricsEndpoint(network, address, cert, key string, me *http.ServeMux)
 	return nil
 }
 
+type certReloader struct {
```

Comment:
what would you think about moving this to a separate file? server/server.go is pretty big already

Reply:
maybe server/metrics.go?

Reply:
I'd like to move the whole metrics part into another file/struct. I can follow up on this if you want. Or should I do it within this PR?

Reply:
Sounds good, moved everything over there.